data/docs/alerts-management/terraform-provider-signoz.mdx

```diff
@@ -0,0 +1,296 @@
+---
+date: 2024-12-05
+id: terraform-provider-signoz
+title: Terraform Provider
+---
+# Creating SigNoz Alert using Terraform 
+
+The configuration of alerts for a monitoring target is the very definition of what the team perceives as a “problem” and should be managed as properly as the application code.
+By defining SigNoz alerts as Terraform code, it is possible to manage changes to the alert settings.
+
+## Overview
+
+SigNoz alert notifies you of the change in the state of your service.
+Although getting alerts that work wisely often requires complex configuration, Terraform 'signoz' provider will help you manage SigNoz alerts as reviewable codes.
+
+This document will show you how to:
+- configure the provider
+- create a new alert on SigNoz Cloud
+- start managing preexistent alerts using Terraform
+
+## Prerequisites
+
+- Terraform
+  - version of Terraform runtime: [v0.13.0](https://github.com/hashicorp/terraform/releases/tag/v0.13.0) or later
+  - version of signoz provider: [0.0.4](https://github.com/SigNoz/terraform-provider-signoz/releases/tag/v0.0.4) or later
+- SigNoz Cloud
+  - endpoint URL
+  - access token
+
+## Create SigNoz Alert using Terraform
+
+### Step 1: Configure Terraform environment
+
+Create `.tf` file with an arbitrary name (`main.tf` is commonly used) and add the following content:
+```hcl
+terraform {
+  required_version = ">= 0.13"
+}
+```
+
+### Step 2: Install Terraform sinoz Provider
+
+Add specification block for the signoz provider as follows:
+```diff
+terraform {
+  required_version = ">= 0.13"
+
++ required_providers {
++   signoz = {
++     source = "Signoz/signoz"
++     version ~> "0.0.4"
++   }
++ }
+}
+```
+
+Running `terraform init` will create `.terraform.lock.hcl`.
+
+### Step 3: Configure sinoz Provider
+
+Add `provider {}` block to the `.tf` file as follows:
+```hcl
+provider "signoz" {}
+```
+
+By setting the environmental variables below, the signoz provider will automatically read these variables and authorize with your SigNoz Cloud account:
+``` bash
+SIGNOZ_ACCESS_TOKEN="your-access-token"
+SIGNOZ_ENDPOINT="http://your-signoz-endpoint:3301"
+```
```

Comment:
SigNoz Endpoint would look something like: `https://xyz.us.signoz.cloud`.

Reply:
thank you!
corrected: e5f0888c